pkg/apis/mpi/validation/validation.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+
+	commonv1 "github.com/kubeflow/common/pkg/apis/common/v1"
+
+	mpiv1 "github.com/kubeflow/training-operator/pkg/apis/mpi/v1"
+)
+
+func ValidateV1MpiJobSpec(c *mpiv1.MPIJobSpec) error {
+	if c.MPIReplicaSpecs == nil {
+		return fmt.Errorf("MPIReplicaSpecs is not valid")
+	}
+	launcherExists := false
+	for rType, value := range c.MPIReplicaSpecs {
+		if value == nil || len(value.Template.Spec.Containers) == 0 {
+			return fmt.Errorf("MPIReplicaSpecs is not valid: containers definition expected in %v", rType)
+		}
+		// Make sure the replica type is valid.
+		validReplicaTypes := []commonv1.ReplicaType{mpiv1.MPIReplicaTypeLauncher, mpiv1.MPIReplicaTypeWorker}
+
+		isValidReplicaType := false
+		for _, t := range validReplicaTypes {
+			if t == rType {
+				isValidReplicaType = true
+				break
+			}
+		}
+
+		if !isValidReplicaType {
+			return fmt.Errorf("MPIReplicaType is %v but must be one of %v", rType, validReplicaTypes)
+		}
+
+		//Make sure the image is defined in the container
+		//defaultContainerPresent := false
+		for _, container := range value.Template.Spec.Containers {
+			if container.Image == "" {
+				msg := fmt.Sprintf("MPIReplicaSpec is not valid: Image is undefined in the container of %v", rType)
+				return fmt.Errorf(msg)
+			}
+			// if container.Name == mpiv1.DefaultContainerName {
```

Comment:
Should we remove these code?

Reply:
@alculquicondor We will move the unit tests in the next pr.

Reply:
Please remove this commented code if it's not actually needed.

Reply:
@hackerboy01 Can you address @alculquicondor 's comments?